File: Practica1/src/practica1/FactoriaMontana.java
package practica1;

import java.util.ArrayList;

public class FactoriaMontana implements FactoriaCarreraYBicicleta {

    public FactoriaMontana() {
    }
    
        @Override
    public CarreraMontana crearCarrera( int numBicicletas) {
        ArrayList<BicicletaMontana> participantes = new ArrayList<>();
        boolean retirarse = false;
        
        for (int i=0; i<numBicicletas; i++){
            
            if (i%6==0)
                retirarse = true;
            else
                retirarse = false;
            
            participantes.add(crearBicicleta(i, retirarse));
        }
            CarreraMontana carreraCreada = new CarreraMontana(participantes);
            
            return carreraCreada;
    }

    @Override
    public BicicletaMontana crearBicicleta(int numBicicleta, boolean retirarse) {
            BicicletaMontana bicicletaCreada = new BicicletaMontana(numBicicleta,retirarse);
            return bicicletaCreada;
    }

}

File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/FactoriaMontana.rb
require('./CarreraMontana.rb');
require('./BicicletaMontana.rb');
require('./FactoriaCarreraYBicicleta.rb');

public
class FactoriaMontana < FactoriaCarreraYBicicleta

	public
	def crearCarrera(numBicicletas)
		participantes = Array.new
    
    bicicleta = BicicletaMontana.new(0,false)
    
        
    for i in (0..numBicicletas-1)
      bicicleta_aniadir = crearBicicleta(bicicleta) 
      if (i%6==0)
        bicicleta_aniadir.retirarse = true
      else
        bicicleta_aniadir.retirarse = false
      end
      participantes << bicicleta_aniadir
    end
    carreraCreada = CarreraMontana.new(participantes)
    return carreraCreada
	end

	def crearBicicleta(bicicleta)
		return bicicleta.clone
	end

	def initialize()
    super
	end
end
